run_windows.go

```diff
@@ -0,0 +1,527 @@
+		}
+	}
+
+	attrListBuf := make([]byte, attrBufSize)
+	si.lpAttrList = syscall.Handle(unsafe.Pointer(&attrListBuf[0]))
+	// create attr list with console handler
+	r1, _, err = initializeProcThreadAttributeList.Call(
+		uintptr(si.lpAttrList),                // attr list buf
+		1,                                     // list item count
+		0,                                     // dwFlags: reserved, MUST be 0
+		uintptr(unsafe.Pointer(&attrBufSize)), // size of the list
+	)
+	if r1 == 0 {
+		// false
+		return nil, os.NewSyscallError("InitializeProcThreadAttributeList (create)", err)
+	}
+
+	// https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-updateprocthreadattribute
+	r1, _, err = updateProcThreadAttribute.Call(
+		uintptr(si.lpAttrList), // buf list
+		0,                      // dwFlags: reserved, MUST be 0
+		_PROC_THREAD_ATTRIBUTE_PSEUDOCONSOLE,
+		uintptr(consoleHandle),
+		unsafe.Sizeof(consoleHandle),
+		0,
+		0,
+	)
+	if r1 == 0 {
+		// false
+		_, _, _ = deleteProcThreadAttributeList.Call(uintptr(si.lpAttrList))
+		return nil, os.NewSyscallError("UpdateProcThreadAttribute", err)
+	}
+
+	return &si, nil
+}
+
+// copied from os/exec.(*Cmd).Start
+// start starts the specified command but does not wait for it to complete.
+//
+// If Start returns successfully, the c.Process field will be set.
+//
+// The Wait method will return the exit code and release associated resources
+// once the command exits.
+func start(c *cmd, consoleHandle syscall.Handle) error {
+	if c.lookPathErr != nil {
+		c.closeDescriptors(c.closeAfterStart)
+		c.closeDescriptors(c.closeAfterWait)
+		return c.lookPathErr
+	}
+	if runtime.GOOS == "windows" {
+		lp, err := lookExtensions(c.Path, c.Dir)
+		if err != nil {
+			c.closeDescriptors(c.closeAfterStart)
+			c.closeDescriptors(c.closeAfterWait)
+			return err
+		}
+		c.Path = lp
+	}
+	if c.Process != nil {
+		return errors.New("exec: already started")
+	}
+	if c.ctx != nil {
+		select {
+		case <-c.ctx.Done():
+			c.closeDescriptors(c.closeAfterStart)
+			c.closeDescriptors(c.closeAfterWait)
+			return c.ctx.Err()
+		default:
+		}
+	}
+
+	//c.childFiles = make([]*os.File, 0, 3+len(c.ExtraFiles))
+	//type F func() (*os.File, error)
+	//for _, setupFd := range []F{c.stdin, c.stdout, c.stderr} {
+	//	fd, err := setupFd()
+	//	if err != nil {
+	//		c.closeDescriptors(c.closeAfterStart)
+	//		c.closeDescriptors(c.closeAfterWait)
+	//		return err
+	//	}
+	//	c.childFiles = append(c.childFiles, fd)
+	//}
+	//c.childFiles = append(c.childFiles, c.ExtraFiles...)
+
+	envv, err := c.envv()
+	if err != nil {
+		return err
+	}
+
+	c.Process, err = startProcess(c.Path, c.argv(), &os.ProcAttr{
+		Dir:   c.Dir,
+		Files: c.childFiles,
+		Env:   addCriticalEnv(dedupEnv(envv)),
+		Sys:   c.SysProcAttr,
+	}, consoleHandle)
+	if err != nil {
+		c.closeDescriptors(c.closeAfterStart)
+		c.closeDescriptors(c.closeAfterWait)
+		return err
+	}
+
+	c.closeDescriptors(c.closeAfterStart)
```

Comment:
Can't this fail?

Reply:
same as https://github.com/creack/pty/pull/109#discussion_r607920765